### ISSUE TITLE
Do not monkeypatch pyplot

### ISSUE THREAD

**Issue by maxnoe** [fact-project/pyfact#10]
I would like to get rid of this monkey patching of matplotlib.pyplot.

I would go for something like:

``` python
import matplotlib.pyplot as plt
import fact.plotting as factplot
import numpy as np

fig, ax = plt.subplots()
factplot.camera(np.random.normal(size=1440), ax=ax)
```

What do you think? I think this is more clear than monkey patching pyplot.

**Comment by dneise** [comment]
I think there is still, and always will be a lot of people doing something like this, e.g. in jupyter notebooks:

In[1]:

``` python
import matplotlib as plt
%matplotlib inline
import fact.plotting as factplot
import numpy as np
```

In[10]

``` python
# no explicit fig, ax = plt.subplots()
factplot.camera(np.random.normal(size=1440)) # no explicit ax=ax
```

And I think this is important.... can we achieve that ?
Both simply and clearly?

**Comment by dneise** [comment]
Generally I like to get rid of that monkey patching ... definitely yes! Just not at the cost of noise like `ax=ax`

**Comment by maxnoe** [comment]
The same as it is now.

``` python
if ax is None:
    ax = plt.gca()
```

**Comment by dneise** [comment]
This is an API breaking change. Do you know how many users will die, when we do this?

**Comment by maxnoe** [comment]
At most 10, I suppose.  :D

**Comment by maxnoe** [comment]
Fixed by merging #13
